### DISCUSSION TITLE
Deepgram socket connection using Javascript SDK

### DISCUSSION THREAD

**Discussion by chandrakantwinleads** [deepgram/community#928] [General help]
I am trying to build a voice AI assistant using deepgram's speech to text, openai and eleven labs. I am using the Javascript sdk for live transcriptions. I believe it internally opens a websocket to deepgram for streaming audio. Now the socket opens successfully but it remains unresponsive for some time and then finally it disconnects. The metadata that i receive usually has duration 0, channels 0 on disconnection. Most of the times, I am not able to use the websocket for any good but sometimes it does work. In such instances on closing the connection, I get non zero for channels. Please help.

Here is one of the recent metadata on closing the connection:
{
  "type": "Metadata",
  "transaction_key": "deprecated",
  "request_id": "eb9b0424-e461-4c5a-9b48-792ef036ea5e",
  "sha256": "37ed9259bac9a14fc79191d08a1d37a2e2a32178b44452672e31a8c7eff34c36",
  "created": "2024-09-20T04:53:42.969Z",
  "duration": 0,
  "channels": 0
}

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.
